### PR TITLE
[Rework] Add more realistic agricultural solution to produce raw grain

### PR DESCRIPTION
**Pull request by BjoHart**
#### Summary
Content "Winnowing, new method of obtaining raw grain"

#### Purpose of change
The game still lacks a more realistic agricultural solution to produce raw grain.

#### Describe the solution
Add devices to the game to separate the chaff from the seed.
Using old agricultural methods.

New recipes for grain (winnowing lvl 1/2)
- [x] barley
- [x] buckwheat
- [x] wheat
- [x] canola seeds
- [x] dried lentils
- [x] oats

Remove recpies for air blowing
- [x] barley
- [x] buckwheat
- [x] wheat
- [x] canola seeds
- [x] dried lentils
- [x] oats

New tools
- [x]  leather bellow lvl1 (air blowing 1)
- [x]  fanning mill lvl2 (winnowing 2)

New recepies
- [x]  leather bellow lvl1 (air blowing 1)
- [x]  fanning mill lvl2 (winnowing 2)

Rework sieves to have winnowing qualitiy lvl 1
- [x] wicker sive
- [x] makeshift sieve
- [x] sieve
- [x] mesh colander

Rework fan and air dryer (+ winnowing qualitiy lvl 1) (- air blowing quality)
- [x] air dryer
- [x] fan

New quality
- [x] winnowing

Remove quality
- [x] air blowing

#### Describe alternatives you've considered
None.

#### Testing
Done

#### Additional context
http://www.pioneergirl.com/blog/archives/6013
https://en.wikipedia.org/wiki/Bellows
https://en.wikipedia.org/wiki/Winnowing

### CODE REVIEW

**Comment by Karol1223** [comment]
> There is no device that has the "air blowing quality".

I'm not opposing the addition but uh. That's not true.
image

**Comment by BjoHart** [comment]
> > There is no device that has the "air blowing quality".
> 
> I'm not opposing the addition but uh. That's not true. image
> 
> Should you be able to separate chaff and seeds with a hairdryer? Uh. Good question?

I think that would even work.
After all, you only blow away the light parts of the chaff so that the heavy grain remains.

**Comment by PatrikLundell** [comment]
The machine has the advantage over the existing blowers in that it doesn't require electricity, and so would be suitable to use in Innawoods, but as it's based on historical precedent, I don't think there's any need to make it mod specific (and you could make an argument for finding one in an agricultural museum as it's the kind of thing I'd expect to find in one were I ever to visit one).

I'd clarify the description, as it's currently easy to read it to say the machine uses the wind or bedsheet flapping e.g. through "...straw with a breeze that's generated by a hand cranked fan in contrast to the previous method of relying on a natural wind or flapping a bedsheet to generate a breeze."

**Comment by fairyarmadillo** [comment]
> > There is no device that has the "air blowing quality".
> 
> I'm not opposing the addition but uh. That's not true. image
> 
> Should you be able to separate chaff and seeds with a hairdryer? Uh. Good question?

You can use anything, as long as it blows air. Hand fans were used in Japan and all sorts of tools, including waiting for a windy day, were used in the rest of the world. Typically you drop seeds in front of horizontally blowing air at such a distance that the heavier seeds fall straight down while the chaff blows away.

Hairdryers usually have light enough settings for this, and even if it was on full blast you'd just hold it farther away.

**Comment by BjoHart** [comment]
Ok, I think I need to do a bigger round here.
See the updated description.

File: data/json/items/tool/workshop.json [hunk 1]
```diff
@@ -1767,5 +1767,21 @@
...
+    "description": "A machine for separating seed from chaff.  It mechanized the winnowing process, in which the grain was separated from the lighter chaff and straw by blowing a leather bellows.",
+    "weight": "15000 g",
+    "volume": "75 L",
+    "price": 25000,
+    "price_postapoc": 1000,
+    "to_hit": -3,
+    "material": [ "wood" ],
+    "symbol": ";",
+    "color": "brown",
+    "qualities": [ [ "WINNOW", 2 ] ],
```
Comment (Maleclypse) [inline data/json/items/tool/workshop.json:1784]: Are you going to add any recipes that require Winnow 2?

Comment (BjoHart) [reply data/json/items/tool/workshop.json:1784]: The repices for grain have two new recipes for each grain.
One for winnowing 1 with the sieves, and one for winnowing with the new tool fanning mill, as you can see in my pics.

**Comment by fairyarmadillo** [comment]
If you're adding a winnowing quality, you can get rid of air blowing. That was all it was added for and it will reduce recipe bloat.

Winnowing needs to be possible without any tools, so you may have overcomplicated it here. IRL you can just toss threshed grain in the air and the chaff will blow away, even if theres barely any wind.

**Comment by BjoHart** [comment]
> If you're adding a winnowing quality, you can get rid of air blowing. That was all it was added for and it will reduce recipe bloat. Just give winnowing 1 to the fan and hair dryer and remove the air blowing recipes.
> 
> You've also left duplicate comments in the recipe jsons.

In fact, I would like to leave the air blowing quality in.
It is possible to separate the grain from the chaff in this way.
It's only more effective with winnowing, which is what I wanted to show with the improved time savings at lvl 1 with sieving and lvl 2 with the new machine.
I would give even more than 25% savings when winnowing with the sieves. More like 30-35%.

**Comment by fairyarmadillo** [comment]
> > If you're adding a winnowing quality, you can get rid of air blowing. That was all it was added for and it will reduce recipe bloat. Just give winnowing 1 to the fan and hair dryer and remove the air blowing recipes.
> > You've also left duplicate comments in the recipe jsons.
> 
> In fact, I would like to leave the air blowing quality in. It is possible to separate the grain from the chaff in this way. It's only more effective with winnowing, which is what I wanted to show with the improved time savings at lvl 1 with sieving and lvl 2 with the new machine. I would give even more than 25% savings when winnowing with the sieves. More like 30-35%.

I don't think that's appropriate. A sieve is not going to be more or less effective than a desk fan or a hair dryer in any way that you can provide data on, and as the person who added air blowing, I think air blowing should now be removed, as it serves no other purpose.

You now have five recipes for wheat cluttering the crafting menu, and this applies for every single type of grain. It bloats the UI, fails to accurately simulate anything, and confuses the player for no real benefit. Removing air blowing would cut out one recipe for every grain type and is absolutely the way to go here.

**Comment by fairyarmadillo** [comment]
It's a further issue because we would now have two separate and almost-equal tool qualities which only exist to do the same somewhat obscure job, which is neither friendly to the player nor future contributors, and provides no benefit in return.

**Comment by BjoHart** [comment]
Ok, if this is your will, i will remove the air blowing with this rework tomorrow.

**Pull request merged**